addons/lunch/models/lunch.py

```diff
@@ -0,0 +1,296 @@
+
+    # such as PIZZA, SANDWICH, PASTA, CHINESE, BURGER, ...
+    name = fields.Char('Category', required=True)
+
+
+class lunch_cashmove(models.Model):
+    """
+    lunch cashmove => order or payment
+    """
+    _name = 'lunch.cashmove'
+    _description = 'lunch cashmove'
+
+    @api.multi
+    def name_get(self):
+        return [(cashmove.id, _("Lunch Cashmove #%d") % cashmove.id) for cashmove in self]
+
+    user_id = fields.Many2one('res.users', 'User Name', required=True,
+                              default=lambda self: self.env.uid)
+
+    date = fields.Date('Date', required=True, default=fields.Date.context_today)
+
+    amount = fields.Float('Amount', required=True) # Can be positive or negative
+
+    description = fields.Text('Description') # Can be an order or a payment
+
+    order_id = fields.Many2one('lunch.order.line', 'Order', ondelete='cascade')
+
+    state = fields.Selection([('order', 'Order'), ('payment', 'Payment')],
+                             'Is an order or a Payment', default='payment')
+
+
+class lunch_alert(models.Model):
+    """
+    lunch alert
+    """
+    _name = 'lunch.alert'
+    _description = 'Lunch Alert'
+
+    @api.multi
+    def name_get(self):
+        return [(alert.id, _("Alert #%d") % alert.id) for alert in self]
+
+    @api.model
+    def get_alert_message(self):
+        """
+        This method check if the alert can be displayed today
+        if alert type is specific : compare specific_day(date) with today's date
+        if alert type is week : check today is set as alert (checkbox true) eg. self['monday']
+        if alert type is day : True
+        return : Message if can_display_alert is True else False
+        """
+
+        can_display_alert = {
+            'specific': (self.specific_day == fields.Date.context_today(self)),
+            'week': self[datetime.datetime.now().strftime('%A').lower()],
+            'days': True
+        }
+
+        if can_display_alert[self.alert_type]:
+            mynow = fields.Datetime.context_timestamp(self, datetime.datetime.now())
+            hour_to = int(self.end_hour)
+            min_to = int((self.end_hour - hour_to) * 60)
+            to_alert = datetime.time(hour_to, min_to)
+            hour_from = int(self.start_hour)
+            min_from = int((self.start_hour - hour_from) * 60)
+            from_alert = datetime.time(hour_from, min_from)
+            if from_alert <= mynow.time() <= to_alert:
+                return self.message
+        return False
+
+    message = fields.Text('Message', required=True)
+
+    alert_type = fields.Selection([('specific', 'Specific Day'),
+                                   ('week', 'Every Week'),
+                                   ('days', 'Every Day')],
+                                  string='Recurrency', required=True, select=True, default='specific')
+
+    specific_day = fields.Date('Day', default=fields.Date.context_today)
+
+    monday = fields.Boolean('Monday')
```

Comment:
Just to be pythonic, I would use integers, to easily use isoweekday (https://docs.python.org/2/library/datetime.html#datetime.date.isoweekday) when having to perform some date computation.